src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/AdditionalConditionUploadDetail.kt

```diff
@@ -0,0 +1,53 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity
+
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.Lob
+import javax.persistence.Table
+import javax.validation.constraints.NotNull
+
+@Entity
+@Table(name = "additional_condition_upload_detail")
+data class AdditionalConditionUploadDetail(
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @NotNull
+  val id: Long = -1,
+
+  @NotNull
+  val licenceId: Long,
+
+  @NotNull
+  val additionalConditionId: Long,
+
+  @Lob
+  @Column(name = "original_data", columnDefinition = "BLOB")
+  val originalData: ByteArray? = null,
+
+  @Lob
+  @Column(name = "full_size_image", columnDefinition = "BLOB")
+  val fullSizeImage: ByteArray? = null,
+) {
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (javaClass != other?.javaClass) return false
+
+    other as AdditionalConditionUploadDetail
```

Comment:
Is this line needed as the above line will kick us out of here if `other` isnt of the correct type?

Reply:
Its generated code. The only reason for the equals and hashcode functions is remove the IntelliJ warnings. When a class contains Blobs it suggests to override the hashcode, equals and toString functions to avoid the potentially long binary columns. They're just generated to ignore the the BLOB columns.